src/qtism/runtime/tests/RouteItem.php

```diff
@@ -426,4 +450,69 @@ public function getTimeLimits($excludeItem = false): RouteTimeLimitsCollection
 
         return $timeLimits;
     }
+
+    public function getEffectiveBranchRules(): BranchRuleCollection
+    {
+        if ($this->getBranchRules()->count() > 0) {
+            return $this->getBranchRules();
+        }
+
+        $sectionBranchRules = $this->getEffectiveSectionBranchRules();
+
+        if ($sectionBranchRules === null || $sectionBranchRules->count() > 0) {
+            return $sectionBranchRules ?? new BranchRuleCollection();
+        }
```

Comment:
suggestion: Found this a bit confusing, I'd split the two conditions.

```suggestion
        if( 
            ($sectionBranchRules === null) // or maybe if (!($sectionBranchRules instanceof BranchRuleCollection))
            || (count($sectionBranchRules) == 0) // BranchRuleCollection implements Countable
        ) { 
            return new BranchRuleCollection();
        }
        
        if (count($sectionBranchRules)) > 0) { // BranchRuleCollection implements Countable
            return $sectionBranchRules;
        }
        
        // else: No section branch rules, fallthough to test branch runes for the test part
```

Reply:
@hectoras IMO extra parentheses can also be confusing, on your example, instead of: 

```
if( 
            ($sectionBranchRules === null) // or maybe if (!($sectionBranchRules instanceof BranchRuleCollection))
            || (count($sectionBranchRules) == 0) // BranchRuleCollection implements Countable
        ) { 
            return new BranchRuleCollection();
        }
```

it can simply be:

```
if ($sectionBranchRules === null || count($sectionBranchRules) === 0) { 
    return new BranchRuleCollection();
}
```

Reply:
In this case I would personally make the method `getEffectiveSectionBranchRules` to always return a `BranchRuleCollection` and never null. Would not it solve the problem? 

And I do not see much sense on doing `instanceof` in a method that has specific return type already as a `BranchRuleCollection`.

Reply:
Done: https://github.com/oat-sa/qti-sdk/compare/12bc4edc855ef96737f9c4c0f1bb68d298c85e9d...045e56cea97d32d51d3cf8bca41936c4bfbee66c